runtime/commonMain/src/kotlinx/serialization/internal/Enums.kt

```diff
@@ -28,19 +28,20 @@ public class EnumDescriptor(
         if (this === other) return true
         if (other == null) return false
         if (other !is SerialDescriptor) return false
-        if (other.kind !== UnionKind.ENUM_KIND) return false
+        if (other.kind !== SerialKind.ENUM) return false
         if (serialName != other.serialName) return false
         if (cachedSerialNames() != other.cachedSerialNames()) return false
         return true
     }
 
     override fun toString(): String {
-        return elementNames().joinToString(", ", "$serialName(", ")")
+        return elementNames.joinToString(", ", "$serialName(", ")")
     }
 
     override fun hashCode(): Int {
         var result = serialName.hashCode()
-        result = 31 * result + elementNames().hashCode()
+        val elementsHashCode = elementNames.fold(1) { hash, element -> 31 * hash + element.hashCode() }
```

Comment:
Why different hashing algorithm?

Reply:
It is the same hashing algorithm, it's just a handrolled version of `ArrayList.hashCode`.